source/common/network/connection_impl.h

```diff
@@ -74,7 +74,7 @@ class ConnectionImpl : public virtual Connection,
   uint32_t readBufferLimit() const override { return read_buffer_limit_; }
 
   // Network::BufferSource
-  Buffer::Instance& getReadBuffer() override { return read_buffer_; }
+  Buffer::Instance& getReadBuffer() override { return *read_buffer_; }
   Buffer::Instance& getWriteBuffer() override { return *current_write_buffer_; }
```

Comment:
It's set in the cc file right before fitler_manager_.onWrite and cleared right after.  There's a note that it's a bit of a hack to provide access under the stack of that call :-)